src/main/java/com/iota/iri/network/Node.java

```diff
@@ -742,35 +743,41 @@ public int getReplyQueueSize() {
         return replyQueue.size();
     }
 
-    public class LRUCache<K, V> {
+    public class FIFOCache<K, V> {
 
-        private int capacity;
+        private final int capacity;
+        private final double dropRate;
         private LinkedHashMap<K, V> map;
+        private final SecureRandom rnd = new SecureRandom();
 
-        public LRUCache(int capacity) {
+        public FIFOCache(int capacity, double dropRate) {
             this.capacity = capacity;
+            this.dropRate = dropRate;
             this.map = new LinkedHashMap<>();
         }
 
         public V get(K key) {
             V value = this.map.get(key);
             if (value == null) {
-                value = null;
-            } else {
-                this.put(key, value);
+                return null;
+            }
+            if (rnd.nextDouble() < this.dropRate) {
```

Comment:
remove the previous if and replace this with

`if (value != null && rnd.nextDouble() < this.dropRate) {` ?